pkg/messaging/v1/util.go

```diff
@@ -95,9 +95,13 @@ func isPermanentHTTPHeader(hdr string) bool {
 		"Accept-Charset",
 		"Accept-Language",
 		"Accept-Ranges",
-		// "Authorization",
+		// Connection-specific header fields such as Connection and Keep-Alive are prohibited in HTTP/2.
+		// For grpc-dotnet users, Kestrel server throws the exception if connection header is given.
+		"Connection",
 		"Cache-Control",
 		"Content-Type",
+		// Content-Length is http 1.1 specific.
```

Comment:
```suggestion
		// Remove content-length header since it represents http1.1 payload size,
		// not the sum of the h2 DATA frame payload lengths. See https://httpwg.org/specs/rfc7540.html#malformed
```